src/jit/gentree.h

```diff
@@ -5301,86 +5301,146 @@ struct GenTreeILOffset : public GenTree
+    {
+        return m_treeRoot;
+    }
 
-    Statement* gtNext;
-    Statement* gtPrev;
+    GenTree** GetTreeRootPointer()
+    {
+        return &m_treeRoot;
+    }
 
-    bool compilerAdded;
+    void SetTreeRoot(GenTree* treeRoot)
+    {
+        m_treeRoot = treeRoot;
+    }
 
-    Statement* GetNextStmt()
+    GenTree* GetTreeList() const
     {
-        if (gtNext == nullptr)
-        {
-            return nullptr;
-        }
-        else
-        {
-            return gtNext;
-        }
+        return m_treeList;
     }
 
-    Statement* getPrevStmt()
+    void SetTreeList(GenTree* treeHead)
     {
-        if (gtPrev == nullptr)
-        {
-            return nullptr;
-        }
-        else
-        {
-            return gtPrev;
-        }
+        m_treeList = treeHead;
     }
 
-    Statement(GenTree* expr, IL_OFFSETX offset DEBUGARG(unsigned stmtID))
-        : gtStmtExpr(expr)
-        , gtStmtList(nullptr)
-        , gtInlineContext(nullptr)
-        , gtStmtILoffsx(offset)
-#ifdef DEBUG
-        , gtStmtLastILoffs(BAD_IL_OFFSET)
-        , m_stmtID(stmtID)
-#endif
-        , gtNext(nullptr)
-        , gtPrev(nullptr)
-        , compilerAdded(false)
+    InlineContext* GetInlineContext() const
     {
+        return m_inlineContext;
     }
 
-    bool IsPhiDefnStmt()
+    void SetInlineContext(InlineContext* inlineContext)
     {
-        return gtStmtExpr->IsPhiDefn();
+        m_inlineContext = inlineContext;
     }
 
-    unsigned char GetCostSz() const
+    IL_OFFSETX GetILOffsetX() const
     {
-        return gtStmtExpr->GetCostSz();
+        return m_ILOffsetX;
     }
 
-    unsigned char GetCostEx() const
+    void SetILOffsetX(IL_OFFSETX offsetX)
     {
-        return gtStmtExpr->GetCostEx();
+        m_ILOffsetX = offsetX;
     }
 
 #ifdef DEBUG
+
+    IL_OFFSET GetLastILOffset() const
+    {
+        return m_lastILOffset;
+    }
+
+    void SetLastILOffset(IL_OFFSET lastILOffset)
+    {
+        m_lastILOffset = lastILOffset;
+    }
+
     unsigned GetID() const
     {
         return m_stmtID;
     }
+#endif // DEBUG
+
+    Statement* GetNextStmt() const
+    {
+        return m_next;
+    }
+
+    void SetNextStmt(Statement* nextStmt)
+    {
+        m_next = nextStmt;
+    }
+
+    Statement* GetPrevStmt() const
+    {
+        return m_prev;
+    }
+
+    void SetPrevStmt(Statement* prevStmt)
+    {
+        m_prev = prevStmt;
+    }
+
+    bool IsCompilerAdded() const
+    {
+        return m_compilerAdded;
+    }
+
+    void SetCompilerAdded()
+    {
+        m_compilerAdded = true;
+    }
+
+    bool IsPhiDefnStmt() const
+    {
+        return m_treeRoot->IsPhiDefn();
+    }
+
+    unsigned char GetCostSz() const
+    {
+        return m_treeRoot->GetCostSz();
+    }
+
+    unsigned char GetCostEx() const
+    {
+        return m_treeRoot->GetCostEx();
+    }
+
+private:
+    GenTree* m_treeRoot; // The root of the expression tree.
+    GenTree* m_treeList; // The tree list head (for forward walks).
```

Comment:
While you're here, it would be good to slightly expand these comments so that it's clearer what the distinction is, and that `m_treeList` will be null until we've set the sequencing of the nodes. It might be worth calling out that the root is the last thing evaluated, while the tree list head is the first. These things are possibly not obvious to someone not used to working with the JIT IR.